nautobot/dcim/tests/test_filters.py

```diff
@@ -663,58 +663,83 @@ class RegionTestCase(FilterTestCases.NameSlugFilterTestCase):
-        Region.objects.create(name="Region 2B", slug="region-2b", parent=parent_regions[1])
-        Region.objects.create(name="Region 3A", slug="region-3a", parent=parent_regions[2])
-        Region.objects.create(name="Region 3B", slug="region-3b", parent=parent_regions[2])
+        cls.parent_regions = list(Region.objects.filter(children__isnull=False)[:3])
+        cls.child_regions = list(Region.objects.filter(parent__in=cls.parent_regions)[:3])
 
     def test_description(self):
-        params = {"description": ["A", "B"]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
+        regions = Region.objects.filter(description__isnull=False).exclude(description="")[:2]
+        params = {"description": [regions[0].description, regions[1].description]}
+        self.assertEqual(
+            self.filterset(params, self.queryset).qs.count(),
+            self.queryset.filter(description__exact=regions[0].description).count()
+            + self.queryset.filter(description__exact=regions[1].description).count(),
+        )
 
     def test_parent(self):
-        parent_regions = Region.objects.filter(parent__isnull=True)[:2]
         with self.subTest():
-            params = {"parent_id": [parent_regions[0].pk, parent_regions[1].pk]}
-            self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
+            params = {"parent_id": [self.parent_regions[0].pk, self.parent_regions[1].pk]}
+            self.assertEqual(
+                self.filterset(params, self.queryset).qs.count(),
+                self.queryset.filter(parent__in=[self.parent_regions[0].pk, self.parent_regions[1].pk]).count(),
+            )
         with self.subTest():
-            params = {"parent": [parent_regions[0].slug, parent_regions[1].slug]}
-            self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
+            params = {"parent": [self.parent_regions[0].slug, self.parent_regions[1].slug]}
+            self.assertEqual(
+                self.filterset(params, self.queryset).qs.count(),
+                self.queryset.filter(
+                    parent__slug__in=[self.parent_regions[0].slug, self.parent_regions[1].slug]
+                ).count(),
+            )
 
     def test_children(self):
-        child_region_1a = Region.objects.get(slug="region-1a")
-        child_region_1b = Region.objects.get(slug="region-1b")
-        child_region_2a = Region.objects.get(slug="region-2a")
         with self.subTest():
-            params = {"children": [child_region_1a.pk, child_region_1b.slug]}
-            self.assertEqual(self.filterset(params, self.queryset).qs.count(), 1)
+            params = {"children": [self.child_regions[0].pk, self.child_regions[1].slug]}
+            self.assertEqual(
+                self.filterset(params, self.queryset).qs.count(),
+                self.queryset.filter(children__in=[self.child_regions[0].pk, self.child_regions[1].pk]).count(),
+            )
```

Comment:
This was testing 2 children with the same parent. It would need to be rewritten if we want to recreate that test with randomized fixtures. I would probably just create a new region for this test.